src/content/docs/en/reference/cli-reference.mdx

```diff
@@ -492,11 +506,16 @@ The inline config passed in this object will take highest priority when merging
 
 <p>
 
-**Type:** `"development" | "production"`<br />
-**Default:** `"development"` when running `astro dev`, `"production"` when running `astro build`
+**Type:** `string`<br />
+**Default:** `"development"` when running `astro dev`, `"production"` when running `astro build`<br />
+<Since v="5.0.0" />
 </p>
 
-The mode used when building your site to generate either "development" or "production" code.
+The mode used when developing or building your site (e.g. `"production"`, `"testing"`).
+
+This value is passed to Vite using [the `--mode` flag](#--mode-string) when the `astro build` or `astro dev` commands are run to determine the value of `import.meta.env.MODE`. This also determines which `.env` files are loaded, and therefore the values of `astro:env`. See the [environment variables page](/en/guides/environment-variables/) for more details.
```

Comment:
```suggestion
This value is passed to Vite using [the `--mode` flag](#--mode-string) when the `astro build` or `astro dev` commands are run to determine the value of `import.meta.env.MODE`.

This also determines which `.env` file is loaded (e.g. `env.production`, `env.testing`, and therefore the values of `astro:env`. See the [environment variables page](/en/guides/environment-variables/) for more details.

Note that changing the `mode` does not change the kind of code transform handled by Vite and Astro:

- In `astro dev`, Astro will transform code with debug information.
- In `astro build`, Astro will transform code with the most optimized output and removes debug information.
- In `astro build --devOutput` (new flag), Astro will transform code with debug information like in `astro dev`. 

This enables various usecases like:

```bash
# Run the dev server connected to a "staging" API
astro dev --mode staging

# Build a site that connects to a "staging" API
astro build --mode staging

# Build a site that connects to a "production" API with additional debug information
astro build --devOutput

# Build a site that connects to a "testing" API
astro build --mode testing
```
``````
I really like the information from the changeset. Do you think it's helpful here, or too much?
```

Reply:
i think it can be really useful to have it in the docs!

Reply:
lol, I must have really liked it! Serves me right for editing over two weeks :laughing: